src/Components/FakeTargetsOutline.fs

```diff
@@ -0,0 +1,402 @@
+            setInFsharpActivity inFsharpActivity
+            setInExplorerActivity (not inFsharpActivity)
+
+            if inFsharpActivity then "fake.targetsOutlineInActivity" else "fake.targetsOutline"
+
+    let setShowCodeOutline = Context.cachedSetter<bool> "fake.targetsOutline.show"
+
+    let inline private isFsharpFile (doc : TextDocument) =
+        match doc with
+        | Document.FSharp when doc.uri.scheme = "file" -> true
+        | _ -> false
+
+    let private setShowTargetsOutlineForEditor (textEditor : TextEditor) =
+        let newValue =
+            if textEditor <> undefined then
+                if isFsharpFile textEditor.document || ShowInActivity.showInFsharpActivity() then
+                    isEnabledFor textEditor.document.uri
+                else
+                    false
+            else
+                false
+        setShowCodeOutline newValue
+
+    let onDidChangeConfiguration (evt : ConfigurationChangeEvent) =
+        let textEditor = window.activeTextEditor
+        if textEditor <> undefined && evt.affectsConfiguration(configurationKey, textEditor.document.uri) then
+            setShowTargetsOutlineForEditor window.activeTextEditor
+            refresh.fire textEditor.document.uri
+
+    let private onDidChangeActiveTextEditor (textEditor : TextEditor) =
+        setShowTargetsOutlineForEditor textEditor
+        if textEditor = undefined || (not (isFsharpFile textEditor.document)) then
+            reallyRefresh.fire(None)
+
+
+    type [<Pojo>] RequestLaunch =
+        { name : string
+          ``type`` : string
+          request : string
+          preLaunchTask : string option
+          program : string
+          args : string array
+          cwd : string
+          console : string
+          stopAtEntry : bool 
+          justMyCode : bool
+          requireExactSource : bool }
+
+    let activate (context : ExtensionContext) =
+        setShowTargetsOutlineForEditor window.activeTextEditor
+
+        let treeViewId = ShowInActivity.initializeAndGetId ()
+
+        window.onDidChangeActiveTextEditor.Invoke(unbox onDidChangeActiveTextEditor)
+            |> context.subscriptions.Add
+
+        refresh.event.Invoke(fun uri ->
+            if isEnabledFor uri then
+                reallyRefresh.fire(None)
+            createEmpty)
+            |> context.subscriptions.Add
+
+        workspace.onDidChangeConfiguration.Invoke(unbox onDidChangeConfiguration)
+            |> context.subscriptions.Add
+
+        commands.registerCommand("FAKE.targetsOutline.goTo", Func<obj, obj>(fun n ->
+            let m = unbox<Model> n
+            match m.Declaration with
+            | Some decl -> 
+                let line = decl.Line
+                let args =
+                    createObj [
+                        "lineNumber" ==> line - 1
+                        "at" ==> "center"
+                    ]
+
+                vscode.commands.executeCommand("revealLine", args)
```

Comment:
Is there another way to do this, this is the code from the previous F# outline but sometimes it feels broken and doesn't work well?

We can improve that later on.